### PR TITLE
Always retry sendGracefulShutdownToCoordinator, #26214

### PR DESCRIPTION
**Pull request by patriknw**
* I could reproduce the issue locally with debug logging and it's clear that it's a
  timing issue. The GracefulShutdownReq message goes to deadletters and it's not
  retried because the coordinator variable was unset.
* cluster-sharding-shutdown-region phase of CoordinatedShutdown timed out

Refs #26214

### CODE REVIEW

File: akka-cluster-sharding/src/test/scala/akka/cluster/sharding/ClusterShardingInternalsSpec.scala [hunk 1]
```diff
@@ -15,12 +15,26 @@ import org.scalatest.mockito.MockitoSugar
 
 import scala.concurrent.duration._
 
+object ClusterShardingInternalsSpec {
```
Comment (patriknw) [inline akka-cluster-sharding/src/test/scala/akka/cluster/sharding/ClusterShardingInternalsSpec.scala]: changes to this file are unrelated cleanup

**Review by helena — APPROVED** [review]
LGTM

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala [hunk 2]
```diff
@@ -590,16 +590,18 @@ private[akka] class ShardRegion(
...
       if (coordinator.isEmpty)
         register()
       else {
-        sendGracefulShutdownToCoordinator()
         requestShardBufferHomes()
-        tryCompleteGracefulShutdown()
       }
 
+      tryCompleteGracefulShutdown()
+
```
Comment (johanandren) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala]: Was a bit confused first before I realised that `sendGracefulShutdownToCoordinator` only actually does that if there is a graceful shutdown in progress, I think the name is a bit misleading, but I can't think of anything making it more clear otoh.

**Review by johanandren — APPROVED** [review]
LGTM
